test/powershell/Host/Logging.Tests.ps1

```diff
@@ -0,0 +1,191 @@
+    $filename = [Guid]::NewGuid().ToString('N')
+    $fullPath = Join-Path -Path $TestDrive -ChildPath "$filename.config.json"
+
+    $null = $sb.AppendLine('{')
+    $null = $sb.AppendFormat('"LogIdentity": "{0}"', $LogId)
+
+    [string] $channels = [string]::Empty
+    if ($LogChannels -ne $null)
+    {
+        $channels = $LogChannels -join ', '
+    }
+    [string] $keywords = [string]::Empty
+    if ($LogKeywords -ne $null)
+    {
+        $keywords = $LogKeywords -join ', '
+    }
+
+    if ($null -ne $LogLevel)
+    {
+        $null = $sb.AppendLine(',')
+        $null = $sb.AppendFormat('"LogLevel": "{0}"', $LogLevel.ToString())
+    }
+    if ([string]::IsNullOrEmpty($channels) -eq $false)
+    {
+        $null = $sb.AppendLine(',')
+        $null = $sb.AppendFormat('"LogChannels": "{0}"', $channels)
+    }
+    if ([string]::IsNullOrEmpty($keywords) -eq $false)
+    {
+        $null = $sb.AppendLine(',')
+        $null = $sb.AppendFormat('"LogKeywords": "{0}"', $keywords)
+    }
+
+    $null = $sb.AppendLine()
+    $null = $sb.AppendLine('}')
+
+    $sb.ToString() | Set-Content -Path $fullPath -ErrorAction Stop
+    return $fullPath
+}
+
+class TestSettings
+{
+    [bool] $IsSupportedEnvironment
+    [string] $SysLogFile = [string]::Empty
+
+    TestSettings([bool] $linux, [string]$pwshHome)
+    {
+        $this.IsSupportedEnvironment = ($linux -eq $true)
+
+        if ($this.IsSupportedEnvironment)
+        {
+            if (Test-Path -Path '/var/log/syslog')
+            {
+                $this.SysLogFile = '/var/log/syslog'
+            }
+            elseif (Test-Path -Path '/var/log/messages')
+            {
+                $this.SysLogFile = '/var/log/messages'
+            }
+            else
+            {
+                # TODO: Look into journalctl and other variations.
+                Write-Warning -Message 'Unsupported Linux syslog configuration.'
+                $this.IsSupportedEnvironment = $false
+            }
+        }
+
+        if ($this.IsSupportedEnvironment)
+        {
+            $this.IsSupportedEnvironment = Test-IsElevated
+        }
+    }
+}
+
+Describe 'Basic SysLog tests on Linux' -Tag 'CI' {
+
+    BeforeAll {
+        $Settings = [TestSettings]::new($IsLinux -eq $true, $PSHome)
+        $powershell = Join-Path -Path $PSHome -ChildPath 'pwsh'
+    }
+
+    It 'Verifies basic logging with no customizations' -Skip:(!$Settings.IsSupportedEnvironment) {
+        # generate a unique log application id
+        $logId = [Guid]::NewGuid().ToString('N')
+        $now = [DateTime]::Now
+
+        $configFile = WriteLogSettings -LogId $logId
+        & $powershell -NoProfile -SettingsFile $configFile -Command '$env:PSModulePath | out-null'
+
+        # Get log entries from the last 100 that match our id and are after the time we launched Powershell
+        $items = Get-PSSysLog -Path $Settings.SyslogFile -Id $logId -After $now -Tail 100 -Verbose -TotalCount 3
```

Comment:
there's an assumption that there aren't other PowerShell sessions executing and logging, is that correct? Would it be better to look for a specific PID or something?

Reply:
The generated logId ([guid]::new) uniquely identifies all log output from the powershell process. I don't need to rely on the pid.